bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/Client.java

```diff
@@ -0,0 +1,150 @@
+
+                String currency = timeSeriesElement.getElementsByTagName("currency_Unit.name").item(0).getTextContent();
+                String measureUnit = timeSeriesElement.getElementsByTagName("price_Measure_Unit.name").item(0)
+                        .getTextContent();
+
+                NodeList listOfTimeInterval = timeSeriesElement.getElementsByTagName("timeInterval");
+                Node startTimeNode = listOfTimeInterval.item(0);
+                Element startTimeElement = (Element) startTimeNode;
+                String startTime = startTimeElement.getElementsByTagName("start").item(0).getTextContent();
+                ZonedDateTime zonedStartTime = ZonedDateTime.parse(startTime);
+
+                String resolution = timeSeriesElement.getElementsByTagName("resolution").item(0).getTextContent();
+
+                logger.debug("\"timeSeries\" node: {}/{} with start time: {} and resolution {}", (i + 1),
+                        listOfTimeSeries.getLength(), zonedStartTime, resolution);
+
+                NodeList listOfPoints = timeSeriesElement.getElementsByTagName("Point");
+
+                for (int p = 0; p < listOfPoints.getLength() && resolution.equalsIgnoreCase("PT60M"); p++) {
+                    Node pointNode = listOfPoints.item(p);
+                    if (pointNode.getNodeType() == Node.ELEMENT_NODE) {
+                        Element pointElement = (Element) pointNode;
+                        ZonedDateTime timeStamp = calculateDateTime(zonedStartTime, p, resolution);
+                        String price = pointElement.getElementsByTagName("price.amount").item(0).getTextContent();
+                        Double priceAsDouble = Double.parseDouble(price);
+
+                        EntsoeTimeSerie t = new EntsoeTimeSerie(currency, measureUnit, priceAsDouble, timeStamp);
+                        responseList.add(t);
+                        logger.trace("\"Point\" node: {}/{} with values: {} - {} {}/{}", (p + 1),
+                                listOfPoints.getLength(), timeStamp, priceAsDouble, currency, measureUnit);
+                    }
+                }
+            }
+        }
+        return responseList;
+    }
+
+    private ZonedDateTime calculateDateTime(ZonedDateTime start, int iteration, String resolution)
+            throws EntsoeResponseException {
+        switch (resolution) {
+            case "PT15M":
+                return start.plusMinutes(iteration * 15);
+            case "PT30M":
+                return start.plusMinutes(iteration * 30);
+            case "PT60M":
+                return start.plusMinutes(iteration * 60);
+            case "P1D":
+                return start.plusDays(iteration * 1);
+            case "P1M":
+                return start.plusMonths(iteration * 1);
+            case "P1Y":
+                return start.plusYears(iteration * 1);
+            default:
+                throw new EntsoeResponseException("Unknown resolution: " + resolution);
```

Comment:
Could be simplified to something like:
```suggestion
        try {
            return start.plus(Duration.parse(resolution).multipliedBy(iteration));
        } catch (DateTimeParseException e) {
            throw new EntsoeResponseException("Unknown resolution: " + resolution);
        }
```